pkg/collector/corechecks/net/wlan/wlan_windows.go

```diff
@@ -0,0 +1,696 @@
+// 	usBeaconPeriod          uint16
+// 	ullTimestamp            uint64
+// 	ullHostTimestamp        uint64
+// 	usCapabilityInformation uint16
+// 	ulChCenterFrequency     uint32
+// 	wlanRateSet             [256]byte
+// 	ulIeOffset              uint32
+// 	ulIeSize                uint32
+// }
+
+// type WLAN_BSS_LIST struct {
+// 	dwTotalSize     uint32
+// 	dwNumberOfItems uint32
+// 	wlanBssEntries  [1]WLAN_BSS_ENTRY
+// }
+
+// type BssInfo struct {
+// 	lrssi               int32
+// 	linkQuality         uint32
+// 	ulChCenterFrequency uint32
+// }
+
+// IP_ADDR_STRING
+// https://learn.microsoft.com/en-us/windows/win32/api/iptypes/ns-iptypes-ip_addr_string
+type IP_ADDR_STRING struct {
+	Next      *IP_ADDR_STRING
+	IpAddress [16]byte
+	IpMask    [16]byte
+	Context   uint32
+}
+
+// IP_ADAPTER_INFO
+// https://learn.microsoft.com/en-us/windows/win32/api/iptypes/ns-iptypes-ip_adapter_info
+type IP_ADAPTER_INFO struct {
+	Next                *IP_ADAPTER_INFO
+	ComboIndex          uint32
+	AdapterName         [MAX_ADAPTER_NAME_LENGTH + 4]byte
+	Description         [MAX_ADAPTER_DESCRIPTION_LENGTH + 4]byte
+	AddressLength       uint32
+	Address             [MAX_ADAPTER_ADDRESS_LENGTH]byte
+	Index               uint32
+	Type                uint32
+	DhcpEnabled         uint32
+	CurrentIpAddress    *IP_ADDR_STRING
+	IpAddressList       IP_ADDR_STRING
+	GatewayList         IP_ADDR_STRING
+	DhcpServer          IP_ADDR_STRING
+	HaveWins            uint32
+	PrimaryWinsServer   IP_ADDR_STRING
+	SecondaryWinsServer IP_ADDR_STRING
+	LeaseObtained       uint32
+	LeaseExpires        uint32
+}
+
+type adapterInfo struct {
+	macAddress string
+	ipAddress  string
+}
+
+type wlanInfo struct {
+	ssid       string
+	bssid      string
+	phy        string
+	auth       string
+	rssi       int32
+	rxRate     float64
+	txRate     float64
+	channel    uint32
+	signal     uint32
+	adapterMac string
+	adapterIp  string
+}
+
+// ------------------------------------------------------
+//
+// # U t i l i t y   f u n c t i o n s
+//
+// ------------------------------------------------------
+// determines the WiFi band based on frequency
+//  Currently not used but may be used in future
+//
+// func formatBand(frequency uint32) string {
+// 	if frequency >= 2412000 && frequency <= 2484000 {
+// 		return "2.4 GHz"
+// 	} else if frequency >= 5180000 && frequency <= 5825000 {
+// 		return "5 GHz"
+// 	} else if frequency >= 5925000 {
+// 		return "6 GHz"
+// 	}
+// 	return "Unknown"
+// }
+
+func formatSSID(dot11Ssid DOT11_SSID) string {
+	ssidLen := dot11Ssid.USSIDLength
+	if ssidLen == 0 {
+		return ""
+	}
+
+	return string(dot11Ssid.UcSSID[:ssidLen])
+}
+
+func formatMacAddress(mac []byte) string {
+	if len(mac) == 6 {
+		return fmt.Sprintf("%02X:%02X:%02X:%02X:%02X:%02X", mac[0], mac[1], mac[2], mac[3], mac[4], mac[5])
+	} else if len(mac) >= 8 {
+		return fmt.Sprintf("%02X:%02X:%02X:%02X:%02X:%02X:%02X:%02X", mac[0], mac[1], mac[2], mac[3], mac[4], mac[5], mac[6], mac[7])
```

Comment:
Are we fine with not having the remaining bytes if we have more than 8 ?

Reply:
It should not be more than 6 bytes. In fact it should be only 6 bytes. I was a bit overzealous. I will simplify the code to report only on "proper" mac address. Thank you.

Reply:
Yeah I actually started reading the wikipedia article as I wasn't sure if it could be something else than 6
I figured this might also handle some different kind of address